packages/components/src/timeline/index.js

```diff
@@ -0,0 +1,106 @@
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
+import moment from 'moment';
+import PropTypes from 'prop-types';
+import { __ } from '@wordpress/i18n';
+
+const Timeline = ( { className, items } ) => {
+	const timelineClassName = classnames( 'woocommerce-timeline', className );
+
+	// Sort all the items reverse chronologically
+	items.sort( function( a, b ) {
+		return a.datetime - b.datetime;
+	} );
+
+	// Group the items into local day collections
+	const days = [];
+
+	items.forEach( function( item ) {
+		const itemLocalDatetime = moment.unix( item.datetime );
+		const dayNum = parseInt( itemLocalDatetime.format( 'YYYYMMDD' ), 10 );
+		if ( ! ( dayNum in days ) ) {
+			days[ dayNum ] = [];
+		}
+		days[ dayNum ].push( item );
+	} );
+
+	if ( 0 === items.length ) {
+		return (
+			<div className={ timelineClassName }>
+				<p className="timeline_no_events">
+					{ __( 'No events to display' ) }
+				</p>
+			</div>
+		);
+	}
+
+	return (
+		<div className={ timelineClassName }>
+			<ul>
+				{ days.map( function( day, dayNum ) {
+					const dayString = moment( dayNum ).format( 'MMMM D, YYYY' );
```

Comment:
Should we consider using localized date formats here? I think the immediate use case might be US only iirc, but would be nice to keep it less locale specific this being a component that will be reused elsewhere.

Reply:
There is now the date setting from site settings:

```js
import { getSetting } from '@woocommerce/wc-admin-settings';
import { format } from '@wordpress/date';

const dateFormat = getSetting( 'dateFormat' );
const date = format( dateFormat, new Date() );
```
As mentioned above, the component shouldn't interact with settings. Instead the format can be passed down as props.

Reply:
@LevinMedia - your design suggested the format given. What say ye?

Reply:
Just pinging @LevinMedia to bring his attention to this.

Reply:
@allendav @reykjalin Yeah if we can localize the date format, that would be good. Thanks!

Reply:
@timmyc @psealock is there a reason to use `format` from `@wordpress/date` over the [moment localization](https://momentjs.com/docs/#/displaying/format/) options?

Also, if the component should not interact with the settings, how do we get the default format? Should we hard code the default format to `F j, Y` (what it is now), or is there a way to get the localized date format without interacting with the settings?

Reply:
>is there a reason to use format from @wordpress/date over the moment localization options?

We decided not to use moment localizations because the file size in substantial.

>how do we get the default format?

Can the component accept a `dateFormat` prop? We don't want to have components tied to settings because then they can't be reused outside of this context.

Reply:
Yeah, a `dateFormat` prop is easy to add, but I’m wondering what the default value of that prop should be.
I don’t think it should be a required prop, and thus we should provide a default value.

Reply:
Fixed in:

- https://github.com/woocommerce/woocommerce-admin/pull/3614/commits/bdea8da421766d67505a6bde345e93fefa56b5f6
- https://github.com/woocommerce/woocommerce-admin/pull/3614/commits/a4ec62031548bb12491f4b4be92df77840af201b
- https://github.com/woocommerce/woocommerce-admin/pull/3614/commits/c0a0303124113aaa67a241c4a744dac3ad393e0e

Default date is set to what @LevinMedia's design show in Figma: `January 1, 2020`.

@psealock @timmyc I think we're ready to ship this, pending an accepted review.